### PR TITLE
Results

### PR DESCRIPTION
**Pull request by riccardocadei**
Change:
- `cre` object: added parameters and ite estimation.
Add:
- Rules Selection description in `summary.cre`.
- `verbose` parameter in `summary.cre`.
Remove:
- `method_params` and `hyper_params` additional parameters in `summary.cre`.

### CODE REVIEW

**Comment by riccardocadei** [comment]
The error is because I have modified the `cre.summary` and `cre.print` arguments. Personally, I think that now it is clearer. (I could leave `verbose` as an implicit, but then it becomes more difficult for the final user to understand where and how to use it).

File: R/cre.R [hunk 1]
```diff
@@ -176,58 +188,75 @@ cre <- function(y, z, X, method_params, hyper_params){
...
-  sd_ite_inf <- ite_list_inf[["sd_ite"]]
-
-  # Generate rules matrix ------------------------------------------------------
+
+  # Estimate ITE
+  if (is.null(ite)) {
+    logger::log_info("Estimating ITE...")
+    ite_list_inf <- estimate_ite(y = y_inf, z = z_inf, X = X_inf,
+                                 ite_method = getElement(method_params,"ite_method_inf"),
+                                 is_y_binary = getElement(method_params,"is_y_binary"),
```
Comment (Naeemkh) [inline R/cre.R:198]: Same here, and the following lines.

File: R/cre.R [hunk 2]
```diff
@@ -176,58 +188,75 @@ cre <- function(y, z, X, method_params, hyper_params){
...
+  results <- list("M" = M,
+                  "CATE" = cate_inf[["summary"]],
+                  "method_params" = method_params,
+                  "hyper_params" = hyper_params,
+                  "ite_pred" = ite_pred)
   attr(results, "class") <- "cre"
 
   # Sensitivity Analysis -------------------------------------------------------
-  #logger::log_info("3 Sensitivity Analysis")
+  #logger::log_info("Starting Sensitivity Analysis")
```
Comment (Naeemkh) [inline R/cre.R:259]: In the starting log comments, please use trailing `...`.
`logger::log_info("Starting Sensitivity Analysis ... ")`

File: R/estimate_ite_bart.R [hunk 3]
```diff
@@ -12,29 +12,20 @@
...
-    set.seed(seed_vals[1])
     est_ps <- estimate_ps(z, X, ps_method)
     X <- cbind(X, est_ps)
   }
 
-
   bart_fit <- bartCause::bartc(as.matrix(y), as.matrix(z), as.matrix(X),
-                               n.samples = 500, n.burn = 500,
-                               seed = seed_vals[2])
+                               n.samples = 500, n.burn = 500)
```
Comment (Naeemkh) [inline R/estimate_ite_bart.R:28]: Please add a note (`@Note`) that `n.samples = 500`, and `n.burn = 500`.

File: R/estimate_ite_poisson.R [hunk 4]
```diff
@@ -5,39 +5,44 @@
...
+#' @param z The treatment vector.
+#' @param X The features matrix.
+#' @param X_names The names of the covariates.
+#' @param offset Name of the covariate to use as offset (i.e. 'x1') to model the
+#' corresponding outcome rate. NULL to model drectly the outcome counts without
+#' offset.
 #'
 #' @return
-#' a vector of ITE estimates
+#' A vector of ITE estimates
```
Comment (Naeemkh) [inline R/estimate_ite_poisson.R:17]: Please add a period at the end of the sentences.

File: R/generate_rules.R [hunk 5]
```diff
@@ -17,51 +17,41 @@
...
   # Random Forest
-  # TO DO: replace splitting criteria
-  set.seed(seed_1)
-  forest <- randomForest::randomForest(x = X, y = ite_std,
-                                                        sampsize = sf * N,
-                                                        replace = replace,
-                                                        ntree = 1,
-                                                        maxnodes = mn,
-                                                        nodesize = node_size)
+  # TO DO: replace splitting criteria enforcing heterogeneity
```
Comment (Naeemkh) [inline R/generate_rules.R:38]: TO DO --> TODO

File: R/print.R [hunk 6]
```diff
@@ -4,16 +4,18 @@
 #' @description
 #' Prints a brief summary of the CRE object
 #'
-#' @param x A CRE object from running the CRE function
-#' @param ... Additional arguments passed to customize the plot.
```
Comment (Naeemkh) [inline R/print.R:8]: Please return back `...`. `print` is a generic function with numerous parameters.

File: R/print.R [hunk 7]
```diff
@@ -4,16 +4,18 @@
...
+#'
 #'
 #' @return
 #' No return value. This function is called for side effects.
 #'
 #' @export
 #'
-print.cre <- function(x, ...) {
-  print(x$CATE)
+print.cre <- function(x, verbose=2) {
```
Comment (Naeemkh) [inline R/print.R:17]: `print.cre <- function(x, verbose = 2, ...)`

File: R/print.R [hunk 8]
```diff
@@ -4,16 +4,18 @@
...
+  summary(x, verbose)
```
Comment (Naeemkh) [inline R/print.R:18]: `summary(x, verbose, ...)`

File: R/print.R [hunk 9]
```diff
@@ -24,85 +26,93 @@ print.cre <- function(x, ...) {
...
+#' @param object A cre object from running the CRE function.
+#' @param verbose Set level of results description details: only results summary
+#' 0, results+parameters summary 1, results+parameters+rules summary (default 2).
 #'
 #' @return
 #' A summary of the CRE object
 #'
 #' @export
-summary.cre <- function(object, ...) {
+summary.cre <- function(object, verbose=2) {
```
Comment (Naeemkh) [inline R/print.R:37]: `summary.cre` <- function(object, verbose=2, ...)`

File: R/print.R [hunk 10]
```diff
@@ -24,85 +26,93 @@ print.cre <- function(x, ...) {
...
 
-  object <- unclass(object)
-  params <- c(...)
```
Comment (Naeemkh) [inline R/print.R:37]: Also return this line back.

File: R/cre.R [hunk 11]
```diff
@@ -112,62 +127,59 @@
 #'
 #' cre_results <- cre(y, z, X, method_params, hyper_params)
 #'}
-cre <- function(y, z, X, method_params, hyper_params){
+cre <- function(y, z, X, method_params=NULL, hyper_params=NULL, ite=NULL) {
```
[the remark this answers is not in the document]
Comment (riccardocadei) [reply R/cre.R:131]: Inside the `cre` function?

File: R/cre.R (hunk 1 above)
Comment (riccardocadei) [reply R/cre.R:198]: only this line: `'%>%' <- magrittr::'%>%'` ?

File: R/print.R (hunk 10 above)
Comment (riccardocadei) [reply R/print.R:37]: Both of them?

File: R/cre.R (hunk 1 above)
Comment (riccardocadei) [reply R/cre.R:198]: Same what?

**Pull request merged**